src/enzyme_ad/jax/Implementations/HLODerivatives.td

```diff
@@ -541,15 +543,77 @@ def GradEinsumConfigRhs : GlobalExpr</*needsprimal*/0, /*needsshadow*/0, [{
 }]>;
 
 // Derivative rules
+def : HLODerivative<"AcosOp", (Op $x), [
+  (Neg (CheckedDiv (DiffeRet), (Sqrt (Sub (HLOConstantFP<"1">), (Pow $x, (HLOConstantFP<"2">))))))
+]>;
+
+def : HLODerivative<"AcoshOp", (Op $x), [
+  (CheckedDiv (DiffeRet), (CheckedMul (Sqrt (Sub (Pow $x, (HLOConstantFP<"2">)), (HLOConstantFP<"1">))), (Sqrt (Add (Pow $x, (HLOConstantFP<"2">)), (HLOConstantFP<"1">)))))
+]>;
+
 def : HLODerivative<"AddOp", (Op $x, $y),
                     [
                       (DiffeRet),
                       (DiffeRet),
                     ]
                   >;
 
+def : HLOInactiveOp<"AfterAllOp">;
+
+def : HLOInactiveOp<"AndOp">;
```

Comment:
Don't mark this as inactive since we want it to catch bithack errors (and activity analysis will be smart enough otherwise).

Reply:
mmm then `ShiftLeft` and `ShiftRight` should also **not** be marked as inactive? since they are usually used in bithacks

Reply:
Yeah I wouldn't do any of the integer ones unless there's a reason they should be zero like round